xtevent/_eventgenvars.ado

```diff
@@ -173,7 +177,7 @@ program define _eventgenvars, rclass
 	loc norever 0
 	loc bounds 0
 	* show a warning message if we don't know treatment time for some units due to missing values in policyvar 
-	if ("`impute'"=="stag" | "`impute'"=="instag") {
+	if ("`impute'"=="stag" | "`impute'"=="instag" | "`w_type'"=="string") {

```

Comment:
@Constantino-Carreto-Romero I understand that this piece of code is because you want the check below to be ran if window is max or balanced. But this is confusing because these are not excluding cases, right? You can have impute be stag AND w_type can be a string.

Is there a better way to structure this check such that the condition here is only over the impute cases, and instead you run the checks in another place if w_type is a string?

Reply:
@jorpppp 
in https://github.com/JMSLab/xtevent/commit/caec36a0c8b136fe639c163a86fdf2f944ff76a3 I restructured the checks for staggered adoption and now they happend only if `impute(stag | instag)` is specified. Requirement of the `impute` option was implemented in https://github.com/JMSLab/xtevent/pull/178#issuecomment-2088869630. If policyvar doesn't follow staggered adoption, first, there will be a warning message saying why the policyvar doesn't follow staggered adoption. Then, an error message will be issue saying that `window(max | balanced)` cannot be used if policyvar doesn't follow staggered adoption.
For instance:
```
*it reverts
use example31, clear
replace z= 0 in 8
cap noi xtevent y eta , panelvar(i) timevar(t) policyvar(z) impute(stag) window(max)
```
generates the following [message](https://github.com/JMSLab/xtevent/assets/36095845/355f5bf8-31b1-4241-9540-e2f04f6b0f6d).